storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4871,8 +4871,7 @@ void ha_rocksdb::setup_field_converters() {
       If hidden pk exists, we skip this check since the field will never be
       part of the hidden pk.
     */
-    if (!has_hidden_pk(table) &&
-        field->part_of_key.is_set(table->s->primary_key)) {
```

Comment:
In some testcase, field->part_of_key was set to 0 even though field was a column in the primary key. So get_storage_type wasn't getting called, which made m_maybe_unpack_info false. This field was a varchar, and since there was no unpack_info provided for unpacking (because m_maybe_unpack_info = false) it caused an assertion error.

The loop below already checks for essentially the same condition in O(m_key_count), so I removed the part_of_key check. I'm not sure how to have an O(1) check without part_of_key though.